documentation/library-reference/source/coloring-stream/index.rst

```diff
@@ -0,0 +1,221 @@
+   let cs = colorize-stream(*standard-output*);
+   format(cs, "%=Error:%= %s", error-text, $text-reset, error-message);
+
+This is implemented by providing an implementation of :gf:`print`
+specialized on :class:`<text-attributes>` and :class:`<coloring-stream>`.
+
+This can be used directly as well:
+
+.. code-block:: dylan
+
+   let error-text = text-attributes(foreground: $color-red);
+   let cs = colorize-stream(*standard-output*);
+   print(error-text, cs);
+   print("Error:", cs);
+   print($text-reset, cs);
+   print(' ', cs);
+   print(error-message, cs);
+
+The COLORING-STREAM module
+**************************
+
+.. class:: <coloring-stream>
+   :abstract:
+
+   :superclasses: :class:`<wrapper-stream>`
+
+   :description:
+
+     :class:`<coloring-stream>` is the abstract wrapper stream
+     that is used to determine whether or not a stream supports
+     color output to avoid having to check at every write.
+
+.. generic-function:: colorize-stream
+
+   :signature: colorize-stream (stream) => (coloring-stream)
+
+   :parameter stream: An instance of :class:`<stream>`.
+   :value coloring-stream: An instance of :class:`<coloring-stream>`.
+
+   :description:
+
+     Wrap a *stream* with an appropriate instance of
+     :class:`<coloring-stream>`. It uses :gf:`stream-supports-color?`
+     to determine whether or not the underlying stream supports
+     color output.
+
+   :example:
+
+     .. code-block:: dylan
+
+        let stream = colorize-stream(*standard-output*);
+
+.. generic-function:: stream-supports-color?
+
+   :signature: stream-supports-color? (stream) => (well?)
+
+   :parameter stream: An instance of :class:`<stream>`.
+   :value well?: An instance of :drm:`<boolean>`.
+
+   :description:
+
+     Return whether or not the underlying stream supports color
+     output.
+
+     This checks that:
+
+     * The stream is a :class:`<file-stream>`.
+     * :gf:`stream-console?` is true. (On Unix, this means that ``isatty``
+       is true for the stream.)
+     * The ``TERM`` environment variable is not ``"dumb"``.
+     * The ``EMACS`` environment variable is not ``"t"``.
+
+Text Attributes
+===============
+
+.. class:: <text-attributes>
+
+   :superclasses: :drm:`<object>`
+
+   :keyword background: An instance of ``false-or(<text-color>)``.
+   :keyword foreground: An instance of ``false-or(<text-color>)``.
+   :keyword intensity: An instance of ``false-or(<text-intensity>)``.
+
+   :description:
+
+     Instances of ``<text-attributes>`` are used for representing the
+     desired text appearance. They can be passed to :gf:`format` when
+     writing to a :class:`<coloring-stream>`.
+
+     *background* and *foreground*, if given, should be one of the color
+     constants like :const:`$color-red`, :const:`$color-green`, etc.
+
+     *intensity*, if given, should be one of :const:`$bright-text-intensity`,
+     :const:`$dim-color-intensity` or :const:`$normal-text-intensity`.
```

Comment:
I fixed this after your comment, but then I decided to rename the constants to ``$dim-intensity`` and so on.